resources/providers/awslib/kms/provider_test.go

```diff
@@ -0,0 +1,158 @@
+type kmsClientMockReturnVals map[string][]mocks
+
+func TestProviderTestSuite(t *testing.T) {
+	s := new(ProviderTestSuite)
+	s.log = logp.NewLogger("cloudbeat_kms_provider_test_suite")
+
+	if err := logp.TestingSetup(); err != nil {
+		t.Error(err)
+	}
+
+	suite.Run(t, s)
+}
+
+func (s *ProviderTestSuite) SetupTest() {}
+
+func (s *ProviderTestSuite) TearDownTest() {}
+
+var keyId = "21c0ba99-3a6c-4f72-8ef8-8118d4804710"
+
+func (s *ProviderTestSuite) TestProvider_DescribeBuckets() {
+	var tests = []struct {
+		name                    string
+		regions                 []string
+		kmsClientMockReturnVals kmsClientMockReturnVals
+		expected                []awslib.AwsResource
+		expectError             bool
+	}{
+		{
+			name: "Should not return any resources when there aren't any keys",
+			kmsClientMockReturnVals: kmsClientMockReturnVals{
+				"ListKeys": {{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{}}, nil}}},
+			},
+			expected:    []awslib.AwsResource(nil),
+			expectError: false,
+			regions:     []string{awslib.DefaultRegion},
+		},
+		{
+			name: "Should not return a resource when key can't be described",
+			kmsClientMockReturnVals: kmsClientMockReturnVals{
+				"ListKeys": {{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{
+					{KeyId: &keyId},
+				}}, nil}}},
+				"DescribeKey": {{{mock.Anything, mock.Anything}, {nil, errors.New("some error")}}},
+			},
+			expected:    []awslib.AwsResource(nil),
+			expectError: false,
+			regions:     []string{awslib.DefaultRegion},
+		},
+		{
+			name: "Should not return a resource when key is not symmetric",
+			kmsClientMockReturnVals: kmsClientMockReturnVals{
+				"ListKeys": {{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{
+					{KeyId: &keyId},
+				}}, nil}}},
+				// TODO: mock.matchBy({keyId}) ?
+				"DescribeKey": {{{mock.Anything, mock.Anything}, {&kmsClient.DescribeKeyOutput{KeyMetadata: &types.KeyMetadata{KeyId: &keyId, KeySpec: "some string"}}, nil}}},
+			},
+			expected:    []awslib.AwsResource(nil),
+			expectError: false,
+			regions:     []string{awslib.DefaultRegion},
+		},
+		{
+			name: "Should not return a resource when key rotation status can't be described",
+			kmsClientMockReturnVals: kmsClientMockReturnVals{
+				"ListKeys": {{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{
+					{KeyId: &keyId},
+				}}, nil}}},
+				"DescribeKey":          {{{mock.Anything, mock.Anything}, {&kmsClient.DescribeKeyOutput{KeyMetadata: &types.KeyMetadata{KeyId: &keyId, KeySpec: types.KeySpecSymmetricDefault}}, nil}}},
+				"GetKeyRotationStatus": {{{mock.Anything, mock.Anything}, {nil, errors.New("some error")}}},
+			},
+			expected:    []awslib.AwsResource(nil),
+			expectError: false,
+			regions:     []string{awslib.DefaultRegion},
+		},
+		{
+			// TODO: add regions
+			// need to change Mock.On to take multiple ListKeys calls
```

Comment:
Will add a test case for this once i figure out the Mock.On thingy. i'll remove this lines before merging/add a test.